src/Sylius/Bundle/AssociationBundle/DependencyInjection/Configuration.php

```diff
@@ -0,0 +1,126 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Bundle\AssociationBundle\DependencyInjection;
+
+use Sylius\Bundle\AssociationBundle\Form\Type\AssociationType;
+use Sylius\Bundle\AssociationBundle\Form\Type\AssociationTypeType;
+use Sylius\Bundle\ResourceBundle\Controller\ResourceController;
+use Sylius\Bundle\ResourceBundle\Form\Type\ResourceChoiceType;
+use Sylius\Bundle\ResourceBundle\SyliusResourceBundle;
+use Sylius\Component\Association\Model\AssociationType as AssociationTypeModel;
+use Sylius\Component\Resource\Factory\Factory;
+use Symfony\Component\Config\Definition\Builder\ArrayNodeDefinition;
+use Symfony\Component\Config\Definition\Builder\TreeBuilder;
+use Symfony\Component\Config\Definition\ConfigurationInterface;
+
+/**
+ * @author Łuksaz Chruściel <lukasz.chrusciel@lakion.com>
+ */
+class Configuration implements ConfigurationInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function getConfigTreeBuilder()
+    {
+        $treeBuilder = new TreeBuilder();
+        $rootNode = $treeBuilder->root('sylius_association');
+
+        $rootNode
+            ->addDefaultsIfNotSet()
+            ->children()
+                ->scalarNode('driver')->defaultValue(SyliusResourceBundle::DRIVER_DOCTRINE_ORM)->end()
+            ->end()
+        ;
+
+        $this->addResourcesSection($rootNode);
+
+        return $treeBuilder;
+    }
+
+    /**
+     * @param ArrayNodeDefinition $node
+     */
+    private function addResourcesSection(ArrayNodeDefinition $node)
+    {
+        $node
+            ->children()
+                ->arrayNode('resources')
+                    ->useAttributeAsKey('name')
+                    ->prototype('array')
+                        ->children()
+                            ->scalarNode('subject')->isRequired()->end()
+                            ->arrayNode('association')
+                                ->isRequired()
+                                ->addDefaultsIfNotSet()
+                                ->children()
+                                    ->arrayNode('classes')
+                                        ->addDefaultsIfNotSet()
+                                        ->children()
+                                            ->scalarNode('model')->isRequired()->end()
+                                            ->scalarNode('controller')->defaultValue(ResourceController::class)->end()
+                                            ->scalarNode('factory')->defaultValue(Factory::class)->end()
```

Comment:
Add `->scalarNode('repository')->cannotBeEmpty()->end()` to allow using own repository implementations.